portafly/src/tests/components/shared/data-list/DataListTable.test.tsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react'
+import { render } from 'tests/custom-render'
+import {
+  useDataListTable,
+  useDataListPagination,
+  useDataListFilters,
+  useDataListBulkActions
+} from 'components'
+import { DataListTable } from 'tests/components/shared/data-list/examples/DataListTable'
+import { fireEvent, within } from '@testing-library/react'
+import { sortable } from '@patternfly/react-table'
+
+jest.mock('components/shared/data-list/DataListContext')
+const useTableMock = useDataListTable as jest.Mock
+const usePaginationMock = useDataListPagination as jest.Mock
+const useFiltersMock = useDataListFilters as jest.Mock
+const useBulkActionsMock = useDataListBulkActions as jest.Mock
+
+const columns = [
```

Comment:
Maybe columns and rows should be declared inside `DataListTable` and being imported from that file instead.

Reply:
Hmmm, not really... the data in columns and rows should hydrate the DataList component from where it's instantiated. Thus any kind of record collection can have its own structure and DataList stays agnostic